test/js/pageobjects/DocumentUpload.js

```diff
@@ -9,7 +20,81 @@ class DocumentUpload extends Base{
       return input
     })()}
 
-    getUploadButton() { return this.$('.onfido-sdk-ui-Uploader-button')}
+    upload(filename) {
+      const input = this.$('.onfido-sdk-ui-CustomFileInput-input')
+      const pathToTestFiles = '../../features/helpers/resources/'
+      const sendKeysToElement = input.sendKeys(path.join(__dirname, pathToTestFiles + filename))
+      return sendKeysToElement
+    }
+
+    copy(lang) { return locale(lang) }
+
+    verifyCrossDeviceUIElements(copy) {
+      const documentUploadScreenCrossDeviceStrings = copy.cross_device.switch_device
+      verifyElementCopy(this.crossDeviceIcon)
+      verifyElementCopy(this.crossDeviceHeader, documentUploadScreenCrossDeviceStrings.header)
+      verifyElementCopy(this.crossDeviceSubMessage, documentUploadScreenCrossDeviceStrings.header.submessage)
+      verifyElementCopy(this.crossDeviceArrow)
+    }
+
+    verifyUploaderIcon() {
+      verifyElementCopy(this.uploaderIcon)
```

Comment:
what copy is this being tested against?

Reply:
It doesn't test the copy here since we have UI elements that don't have copies but with `verifyElementCopy` we check also whether `element.isDisplayed()`.

For elements that don't have copies, we could have a separate function that only checks `element.isDisplayed()` without verifying copies, so it would be clearer.

Reply:
you can just `this.uploaderIcon.isDisplayed()` no point in calling another function.
I'm also intrigued on how it didn't fail? have you confirmed that `verifyElementCopy` can indeed fail?

Reply:
I'm getting

`UnhandledPromiseRejectionWarning: StaleElementReferenceError: stale element reference: element is not attached to the page document` when I try to fail it but the test still passes 😬 

Need to look at how to fail it properly.